### PR TITLE
Fix backup with killed mutations

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Bug Fix

### Changelog entry:
Fix backup if mutations get killed during the backup process.

### CODE REVIEW

**Comment by vitlibar** [comment]
> I still don't understand why we need to backup mutations....

It backups only unfinished mutations. Right now unfinished mutations are stored in the backup but they are not used during RESTORE. I hope we will be able to handle unfinished mutations later during RESTORE somehow.

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -8903,16 +8903,20 @@ void StorageReplicatedMergeTree::backupData(
...
+        if (zookeeper->tryGetChildren(fs::path(zookeeper_path) / "mutations", mutation_ids) == Coordination::Error::ZOK)
         {
-            mutation_infos.emplace_back(
-                IBackupCoordination::MutationInfo{mutation_id, zookeeper->get(fs::path(zookeeper_path) / "mutations" / mutation_id)});
+            std::vector<IBackupCoordination::MutationInfo> mutation_infos;
+            mutation_infos.reserve(mutation_ids.size());
+            for (const auto & mutation_id : mutation_ids)
+            {
+                String mutation;
+                if (zookeeper->tryGet(fs::path(zookeeper_path) / "mutations" / mutation_id, mutation))
```
Comment (alesapin) [inline src/Storages/StorageReplicatedMergeTree.cpp:8915]: But the table can remove some mutations concurrently. Also, I don't understand -- in `/mutations` node, we store 100 (configurable setting) last mutations: both done and undone.

Again, I'd prefer to remove this code.

**Review by alesapin — APPROVED** [review]
If you still want to store this info in a backup -- Ok, let's do it. But it's almost useless from my point of view and even leads to issues (as we see in this PR).

**Comment by vitlibar** [comment]
> If you still want to store this info in a backup -- Ok, let's do it. But it's almost useless from my point of view and even leads to issues (as we see in this PR).

I'd prefer to keep it for now. Maybe later we'll remove it anyway.

**Pull request merged**